src/types/local_date.rs

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /// Create a new object from number of days since 01.01.1970.
+    #[napi]
+    pub fn new_day(value: i32) -> napi::Result<Self> {
+        let date = Ymd::from_days(value.into())
+            .map(|x| (Some(x.year), Some(x.month), Some(x.day)))
+            .unwrap_or((None, None, None));
+        Ok(LocalDateWrapper {
+            value,
+            year: date.0,
+            month: date.1,
+            day: date.2,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        })
+    }
+
+    /// Returns the number of days in a month.
+    fn days_in_month(month: i8, year: i32) -> i8 {
+        match month {
+            1 | 3 | 5 | 7 | 8 | 10 | 12 => 31,
+            4 | 6 | 9 | 11 => 30,
+            2 => {
+                if is_leap_year(year) {
+                    29
+                } else {
+                    28
+                }
+            }
+            _ => 0,
+        }
+    }
+
+    /// Checks if the given date is correct and returns Self.
+    fn check_valid_date(date: Ymd) -> napi::Result<Self> {
+        if !(1..=12).contains(&date.month) {
+            return Err(js_error("Invalid month"));
+        }
+
+        if !(date.day >= 1 && date.day <= Self::days_in_month(date.month, date.year)) {
+            return Err(js_error("Invalid number of day"));
+        }
+
+        let value = date.to_days();
+
+        Ok(LocalDateWrapper {
+            year: Some(date.year),
+            month: Some(date.month),
+            day: Some(date.day),
+            value,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        })
+    }
+
+    #[napi(js_name = "toString")]
+    pub fn to_format(&self) -> String {
+        self.to_string()
+    }
+
+    pub fn get_cql_date(&self) -> CqlDate {
+        CqlDate(((1 << 31) + self.value) as u32)
+    }
+
+    pub fn from_cql_date(date: CqlDate) -> Self {
+        let value: i32 = date.0 as i32 - (1 << 31);
+        let date = Ymd::from_days(value.into())
+            .map(|x| (Some(x.year), Some(x.month), Some(x.day)))
+            .unwrap_or((None, None, None));
+        LocalDateWrapper {
+            value,
+            year: date.0,
+            month: date.1,
+            day: date.2,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        }
+    }
+
+    /// Returns the number of days since 01.01.1970 based on a String representing the date.
+    #[napi]
+    pub fn from_string(value: String) -> napi::Result<i32> {
+        match value.chars().filter(|c| *c == '-').count() {
+            d if d < 2 => match value.parse::<i32>() {
+                Ok(val) => Ok(val),
+                Err(_) => Err(js_error("Invalid format of string")),
+            },
+            2 | 3 => {
+                let lambda = |s: String| -> Result<(i32, i8, i8), ParseIntError> {
+                    let mut cut: usize = 0;
+                    if s.starts_with('-') {
+                        cut = 1;
+                    }
```

Comment:
:recycle: Please avoid needless `mut` in variables. This can be easily made a non-`mut` `let`.